GitVersionCore/GitVersionContext.cs

```diff
@@ -11,12 +11,12 @@ public class GitVersionContext
     {
         readonly Config configuration;
 
-        public GitVersionContext(IRepository repository, Config configuration, bool isForTrackingBranchOnly = true)
-            : this(repository, repository.Head, configuration, isForTrackingBranchOnly)
+        public GitVersionContext(IRepository repository, Config configuration, bool isForTrackingBranchOnly = true, string commitId = null)
```

Comment:
I think this should be on the config with the branch and such? Otherwise looks good